homeassistant/components/flexit_bacnet/sensor.py

```diff
@@ -0,0 +1,189 @@
+    StateType,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    PERCENTAGE,
+    REVOLUTIONS_PER_MINUTE,
+    UnitOfPower,
+    UnitOfTemperature,
+    UnitOfTime,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import FlexitCoordinator
+from .const import DOMAIN
+from .entity import FlexitEntity
+
+
+@dataclass(kw_only=True, frozen=True)
+class FlexitSensorEntityDescription(SensorEntityDescription):
+    """Describes a Flexit sensor entity."""
+
+    value_fn: Callable[[FlexitBACnet], float]
+
+
+SENSOR_TYPES: tuple[FlexitSensorEntityDescription, ...] = (
+    FlexitSensorEntityDescription(
+        key="outside_air_temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        translation_key="outside_air_temperature",
+        value_fn=lambda data: data.outside_air_temperature,
+    ),
+    FlexitSensorEntityDescription(
+        key="supply_air_temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        translation_key="supply_air_temperature",
+        value_fn=lambda data: data.supply_air_temperature,
+    ),
+    FlexitSensorEntityDescription(
+        key="exhaust_air_temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        translation_key="exhaust_air_temperature",
+        value_fn=lambda data: data.exhaust_air_temperature,
+    ),
+    FlexitSensorEntityDescription(
+        key="extract_air_temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        translation_key="extract_air_temperature",
+        value_fn=lambda data: data.extract_air_temperature,
+    ),
+    FlexitSensorEntityDescription(
+        key="room_temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        translation_key="room_temperature",
+        value_fn=lambda data: data.room_temperature,
+    ),
+    FlexitSensorEntityDescription(
+        key="fireplace_ventilation_remaining_duration",
+        device_class=SensorDeviceClass.DURATION,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_unit_of_measurement=UnitOfTime.MINUTES,
+        native_unit_of_measurement=UnitOfTime.MINUTES,
+        translation_key="fireplace_ventilation_remaining_duration",
+        value_fn=lambda data: data.fireplace_ventilation_remaining_duration,
+        suggested_display_precision=0,
+    ),
+    FlexitSensorEntityDescription(
+        key="rapid_ventilation_remaining_duration",
+        device_class=SensorDeviceClass.DURATION,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_unit_of_measurement=UnitOfTime.MINUTES,
```

Comment:
Not sure if having both suggested and native the same has any difference

Reply:
Done